sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py

```diff
@@ -73,7 +73,7 @@ def valid_metrics(cls) -> List[str]:
             "hamming",
             *BOOL_METRICS,
         }
-        return sorted(set(METRIC_MAPPING.keys()) - excluded)
+        return sorted(({"sqeuclidean"} | set(METRIC_MAPPING.keys())) - excluded)
```

Comment:
Should we use `_VALID_METRICS` instead ? https://github.com/scikit-learn/scikit-learn/blob/27d3899432bef52e6706f9112cab86837497e204/sklearn/metrics/pairwise.py#L1622

Reply:
`_VALID_METRICS` and `METRIC_MAPPING.keys()` intersect but none isn't included in the other.

`METRIC_MAPPING.keys()` is the list of `DistanceMetric` within scikit-learn while `_VALID_METRICS` has other distance metrics (like `"sqeuclidean"` that we conceptually support without any `DistanceMetric` via the `Euclidean*` specialisation).

Hence, I think f9beef19ec2207f44000d24d3b2e197266ef8b6c can be reverted and we can implement new `DistanceMetric` in the future (or have our be in common with SciPy's).

What do you think?

Reply:
Here are the metrics in VALID_METRICS that are not in METRIC_MAPPING:
```
    "correlation",
    "cosine",
    "sqeuclidean",
    "yule",
    "nan_euclidean",
```

and here are the ones that are in METRIC_MAPPING but not in VALID_METRICS:
```
    'p'
    'infinity'
    'pyfunc'
```
I agree, let's keep using METRIC_MAPPING for now. At some point it would nice to have more consistency throughout the pairwise module, but let's leave that for later :)